### PR TITLE
Add client for DfmHttpService

### PR DESCRIPTION
**Pull request by qinezh**

### CODE REVIEW

**Comment by vwxyzh** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

File: src/VscPreviewExtension/src/dfmHttpClient.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import axios from 'axios';
+import { AxiosResponse } from 'axios'
+
+import { DfmServiceResult } from './dfmServiceResult';
+
+export class DfmHttpClient {
+    // TODO: make the urlPrefix configurable
+    private urlPrefix = "http://localhost:4001";
+
+    async SendPostRequestAsync(command: String, content: String): Promise<DfmServiceResult> {
```
Comment (hellosnow) [inline src/VscPreviewExtension/src/dfmHttpClient.ts]: Use camelCase for function names: https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines

File: src/VscPreviewExtension/src/dfmService.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+export class DfmService {
+    private static client = new DfmHttpClient();
+
+    static async PreviewAsync(content: String): Promise<DfmServiceResult> {
+        if (!content) {
+            return null;
+        }
+
+        let markup = await DfmService.client.SendPostRequestAsync("preview", content);
```
Comment (hellosnow) [inline src/VscPreviewExtension/src/dfmService.ts]: minor: can not using let and return directly?

File: src/VscPreviewExtension/src/dfmHttpClient.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import axios from 'axios';
+import { AxiosResponse } from 'axios'
```
Comment (hellosnow) [inline src/VscPreviewExtension/src/dfmHttpClient.ts]: can we use import { axios, AxiosResponse } from 'axios'?

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**